extension/app/src/lib/auth.ts

```diff
@@ -60,35 +81,48 @@ export const logout = async (): Promise<boolean> => {
 // Refresh token sends a message to the background script to call the auth0 refresh token endpoint.
 // It updates the stored tokens with the new access token.
 // If the refresh token is invalid, it will call handleLogout.
-export const refreshToken = async (): Promise<StoredTokens | undefined> => {
+export const refreshToken = async (): Promise<
+  Result<StoredTokens, AuthError>
+> => {
   try {
     const tokens = await getStoredTokens();
     if (!tokens) {
-      throw new Error("No tokens found.");
+      return new Err(new AuthError("not_authenticated", "No tokens found."));
     }
     const response = await sendRefreshTokenMessage(tokens.refreshToken);
     if (!response?.accessToken) {
-      throw new Error("No access token received.");
+      return new Err(
+        new AuthError("not_authenticated", "No access token received")
+      );
     }
-    return await saveTokens(response);
+    return new Ok(await saveTokens(response));
   } catch (error) {
     log("Refresh token: unknown error.", error);
     await logout();
+    return new Err(new AuthError("not_authenticated", error?.toString()));
   }
 };
 
 // Fetch me sends a request to the /me route to get the user info.
 const fetchMe = async (
   token: string
-): Promise<{ user: UserTypeWithWorkspaces }> => {
+): Promise<Result<{ user: UserTypeWithWorkspaces }, AuthError>> => {
   const response = await fetch(`${process.env.DUST_DOMAIN}/api/v1/me`, {
     headers: { Authorization: `Bearer ${token}` },
   });
+  const me = await response.json();
   if (!response.ok) {
-    throw new Error(
-      `Failed to fetch /me: ${response.status} ${response.statusText}`
-    );
+    return new Err(new AuthError(me.error.type, me.error.message));
   }
-  const me = await response.json();
-  return me;
+  return new Ok(me);
+};
+
+export const useAuthErrorCheck = (error: any) => {
```

Comment:
Why do we have it twice? (here and in extension/app/src/hooks/useAuthErrorCheck.ts)

Reply:
removed